zerver/openapi/zulip.yaml

```diff
@@ -20149,6 +20155,40 @@ paths:
                       "new": {"direct_members": [10], "direct_subgroups": [11]},
                       "old": 15,
                     }
+                can_send_message_group:
+                  description: |
+                    The set of users who have permission to post in this channel
+                    expressed as an [update to a group-setting value][update-group-setting].
+
+                    Note that a user who is a member of the specified user group must
+                    also [have access](/help/channel-permissions) to the channel in
+                    order to post in the channel.
+
+                    **Changes**: New in Zulip 10.0 (feature level 333). Previously
+                    `stream_post_policy` field used to control the permission to
+                    post in the channel.
+                  type: object
+                  additionalProperties: false
+                  properties:
+                    new:
+                      allOf:
+                        - description: |
+                            The new [group-setting value](/api/group-setting-values) for who would
+                            have the permission to post in this channel.
+                        - $ref: "#/components/schemas/CanSendMessageGroup"
+                    old:
+                      allOf:
+                        - description: |
+                            The expected current [group-setting value](/api/group-setting-values)
+                            for who has the permission to post in this channel.
+                        - $ref: "#/components/schemas/CanSendMessageGroup"
```

Comment:
We can probably deduplicate this whole old/new block into a shared component by just doing "for who has this permission" as the wording.

Reply:
Kept example out of this component as different settings have different `old` values.

Reply:
OK -- maybe worth discussion in "#api documentation" whether that's important, or we can deduplicate further.

Reply:
Since different settings have different default values, the `old` values should be set accordingly for each setting for API tests to pass. Maybe we can skip passing some of the parameters in the request made in tests.